### PR TITLE
fix: try to streamline logic

### PR DESCRIPTION
**Pull request by cavacado**
This pr attempts to reduce the amount of conditional branching done in the interpretation of the fragments.

Could i get another pair of eyes or brain to help me see whether the logic is equivalent

### CODE REVIEW

File: src/fragments/fragments.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const interpretFragments = (fragments: VerificationFragment[]): interpret
 
 export const errorMessageHandling = (fragments: VerificationFragment[]): string[] => {
   const { hashValid, issuedValid, identityValid } = interpretFragments(fragments);
-  const errors = [];
+  let errors = [];
```
Comment (superical) [inline src/fragments/fragments.ts:19]: The changes seem to only use `errors.push`? Then maybe we can still use `const`?

Comment (cavacado) [reply src/fragments/fragments.ts:19]: yea i was testing something out; forgot to change back to `const` will do so

File: src/fragments/fragments.ts [hunk 2]
```diff
@@ -30,21 +30,24 @@ export const errorMessageHandling = (fragments: VerificationFragment[]): string[
     // if the error is because cannot connect to Ethereum, only return this one
     return [TYPES.SERVER_ERROR];
   }
+  if (!issuedValid && !hashValid && !identityValid) {
+    // this error is caused when the document is invalid, only keep this one
+    return [TYPES.INVALID];
+  }
```
Comment (superical) [inline src/fragments/fragments.ts:36]: Notes: Moved this part from nested `if` clause inside `!issuedValid` check to its own clause outside.✅

File: src/fragments/fragments.test.ts [hunk 3]
```diff
@@ -85,9 +85,7 @@ describe("interpretFragments", () => {
 describe("errorMessageHandling", () => {
   it("should return all errors when fragments have multiple errors", () => {
     expect(errorMessageHandling(whenDocumentHashInvalidAndNotIssued as VerificationFragment[])).toStrictEqual([
-      "HASH",
-      "IDENTITY",
-      "ISSUED",
+      "INVALID",
```
Comment (superical) [inline src/fragments/fragments.test.ts:88]: Just thinking if this test is to check that multiple errors will be returned, why do we change it from the 3 errors to a single `INVALID` error?

**Pull request closed without merging**